src/internal/modules/charge-information/pre-handlers.js

```diff
@@ -126,14 +126,31 @@ const loadChargeVersion = async request => {
   }
 };
 
+const flattenAdditionalChargesProperties = ({ additionalCharges, ...element }) => {
+  if (additionalCharges) {
+    const { supportedSource, isSupplyPublicWater } = additionalCharges;
+    element.isAdditionalCharges = true;
+    element.isSupportedSource = !!supportedSource;
+    element.isSupplyPublicWater = isSupplyPublicWater;
+    const { id, name } = supportedSource;
+    if (id) {
```

Comment:
Is the 'if' statement necessary? (Genuine question)

I'm thinking that if 'id' is undefined for example, then you can still set `element.supportedSourceId = id` and it probably wouldn't make any difference

Reply:
I didn't want the property to exist if it was set to undefined.